pkg/controllers/edgeconnect/controller.go

```diff
@@ -158,13 +160,17 @@ func (controller *Controller) reconcileEdgeConnect(ctx context.Context, edgeConn
 		log.Info("status changed, updating EdgeConnect")
 
 		if errClient := controller.updateEdgeConnectStatus(ctx, edgeConnect); errClient != nil {
-			return reconcile.Result{RequeueAfter: fastUpdateInterval}, errors.WithMessagef(errClient, "failed to update EdgeConnect after failure, original error: %s", err)
+			retErr := errors.WithMessagef(errClient, "failed to update EdgeConnect after failure, original error: %s", err)
+			log.Debug("reconcileEdgeConnect error", "error", retErr)
+
+			return reconcile.Result{RequeueAfter: fastUpdateInterval}, retErr
```

Comment:
this will just print the error twice, once via Debug, once by the controller-runtime because you returned the error